src/app/qgsmaptoolchamferfillet.h

```diff
@@ -0,0 +1,126 @@
+    begin                : September 2025
+    copyright            : (C) 2025 by Oslandia
+    email                : benoit dot de dot mezzo at oslandia dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSMAPTOOLCHAMFERFILLET_H
+#define QGSMAPTOOLCHAMFERFILLET_H
+
+#include "qgsmaptooledit.h"
+#include "qgsgeometry.h"
+#include "qgis_app.h"
+#include "ui_qgschamferfilletuserinputwidget.h"
+#include "qgspointlocator.h"
+#include "qgsfeature.h"
+
+class QGridLayout;
+
+class QgsSnapIndicator;
+class QgsDoubleSpinBox;
+class QGraphicsProxyWidget;
+class QgsFeature;
+
+class APP_EXPORT QgsChamferFilletUserWidget : public QWidget, private Ui::QgsChamferFilletUserInputBase
+{
+    Q_OBJECT
+
+  public:
+    explicit QgsChamferFilletUserWidget( QWidget *parent = nullptr );
+
+    void setValue1( double value1 );
+    double value1() const;
+    void setValue2( double value2 );
+    double value2() const;
+    void setMaximumValue1( double maximum );
+    QDoubleSpinBox *editor() const { return mValue1SpinBox; }
+    QString operation() const;
+
+  signals:
+    void distanceEditingFinished( const Qt::KeyboardModifiers &modifiers );
+    void distanceEditingCanceled();
+    void distanceConfigChanged();
+
+  protected:
+    bool eventFilter( QObject *obj, QEvent *ev ) override;
+};
+
+class APP_EXPORT QgsMapToolChamferFillet : public QgsMapToolEdit
+{
+    Q_OBJECT
+  public:
+    QgsMapToolChamferFillet( QgsMapCanvas *canvas );
+    ~QgsMapToolChamferFillet() override;
+
+    void keyPressEvent( QKeyEvent *e ) override;
+    void canvasReleaseEvent( QgsMapMouseEvent *e ) override;
+    void canvasMoveEvent( QgsMapMouseEvent *e ) override;
+
+  private slots:
+    //! Places curve chamfer from the mouse position or from the value entered in the spin box
+    void updateGeometryAndRubberBand( double value1, double value2 );
+
+    void applyOperationFromWidget( Qt::KeyboardModifiers modifiers );
+
+    //! Apply the chamfer either from the spin box or from the mouse event
+    void applyOperation( double value1, double value2, Qt::KeyboardModifiers modifiers );
+
+    void cancel();
+
+  private:
+    //! Rubberband that shows the position of the chamfer curve
+    QgsRubberBand *mRubberBand = nullptr;
+    //! Snapping indicators
+    std::unique_ptr<QgsSnapIndicator> mSnapIndicator;
+
+    //! The layer being manipulated
+    QgsVectorLayer *mSourceLayer = nullptr;
```

Comment:
For safety this should be QPointer

Reply:
ok but as QgsPointLocator::Match::layer() does not return a QPointer what we will gain?